
Accumulated compile fixes. cassert since gcc-3.3 and the template reordering
since gcc-3.4.

  - Juergen G. Sawinsiki <>
  - Rene Rebe <rene@exactcode.de>

--- gsmlib-1.10/gsmlib/gsm_unix_serial.h~	2002-09-22 13:51:10.000000000 +0200
+++ gsmlib-1.10/gsmlib/gsm_unix_serial.h	2004-03-14 04:16:26.000000000 +0100
@@ -13,6 +13,7 @@
 #ifndef GSM_UNIX_SERIAL_H
 #define GSM_UNIX_SERIAL_H
 
+#include <cassert>
 #include <string>
 #include <gsmlib/gsm_error.h>
 #include <gsmlib/gsm_port.h>
--- gsmlib-1.10/gsmlib/gsm_map_key.h~	2002-05-14 21:38:12.000000000 +0200
+++ gsmlib-1.10/gsmlib/gsm_map_key.h	2004-12-29 21:10:45.991761416 +0100
@@ -25,6 +25,16 @@
 
   // wrapper for map key, can access Sortedtore to get sortOrder()
 
+  template <class SortedStore> class MapKey;
+
+  // compare two keys
+  template <class SortedStore>
+    extern bool operator<(const MapKey<SortedStore> &x,
+                          const MapKey<SortedStore> &y);
+  template <class SortedStore>
+    extern bool operator==(const MapKey<SortedStore> &x,
+                           const MapKey<SortedStore> &y);
+
   template <class SortedStore> class MapKey
   {
     SortedStore &_myStore;   // my store
@@ -61,14 +71,6 @@
                        const MapKey<SortedStore> &y);
   };
 
-  // compare two keys
-  template <class SortedStore>
-    extern bool operator<(const MapKey<SortedStore> &x,
-                          const MapKey<SortedStore> &y);
-  template <class SortedStore>
-    extern bool operator==(const MapKey<SortedStore> &x,
-                           const MapKey<SortedStore> &y);
-  
   // MapKey members
   
   template <class SortedStore>
